src/storage/mvcc/reader/point_getter.rs

```diff
@@ -151,15 +155,35 @@ impl<S: Snapshot> PointGetter<S> {
             }
         }
 
+        let mut can_be_cached_by_lock = true;
         match self.isolation_level {
             IsolationLevel::Si => {
                 // Check for locks that signal concurrent writes in Si.
-                self.load_and_check_lock(user_key)?;
+                let check_lock_res =
+                    self.load_and_check_lock(user_key, Some(&mut can_be_cached_by_lock));
+                if let Err(e) = check_lock_res {
+                    *can_be_cached.unwrap() = can_be_cached_by_lock;
+                    return Err(e);
+                }
             }
             IsolationLevel::Rc => {}
         }
 
-        self.load_data(user_key)
+        let mut use_near_seek = false;
+        if can_be_cached_by_lock {
+            if let Some(true) = can_be_cached {
+                self.write_cursor.seek(
+                    &user_key.clone().append_ts(TimeStamp::max()),
+                    &mut self.statistics.write,
+                )?;
+                let current_key = self.write_cursor.key(&mut self.statistics.write);
+                *can_be_cached.unwrap() =
```

Comment:
I think we should compare `current_key` with `user_key` first. If they are different, we don't continue.

Reply:
+1